runtime/fiber.c

```diff
@@ -939,7 +939,7 @@ void caml_free_stack (struct stack_info* stack)
     stack->exception_ptr =
       (void*)(cache[stack->cache_bucket]);
     cache[stack->cache_bucket] = stack;
-#ifdef DEBUG
+#if defined(DEBUG) && defined(STACK_CHECKS_ENABLED)
```

Comment:
Instead of entirely disabling this debug feature, I'd prefer to make it work properly in this case, with something like the following (untested):
```c
char* base;
#if defined(NATIVE_CODE) && !defined(STACK_CHECKS_ENABLED)
  base = Protected_stack_page(stack, caml_plat_pagesize) + caml_plat_pagesize;
#else
  base = (char*)Stack_base(stack);
#endif
memset(base, 0x42, (char*)Stack_high(stack) - base); 
```

Reply:
Got it. I'm not sure why this is enabled even without stack checks while a similar check just a few lines below requires `STACK_CHECKS_ENABLED`, but will do.

Reply:
On second thoughts, let's go with your original implementation instead of my suggestion.

Stacks allocated when stack checks are disabled are huge, and we hope not to actually fault all that memory in but instead rely on lazy OS allocation. It would be a shame to fault it all in just to free it! On programs with lots of effects / systhreads, this could plausibly cause out-of-memory errors in debug builds.

(This means we'll lose the debug check, but oh well)

Reply:
I see, should we merge or there's something else to do?

Reply:
Let's merge!